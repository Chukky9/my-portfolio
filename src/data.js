import javascript from './images/icons/javascript.png';
import laravel from './images/icons/laravel-48.png';
import nodeJs from './images/icons/node-js.png';
import react from './images/icons/react.png';
import vueJs from './images/icons/vue-js.png';

import calculator from './images/calculator.png'
import bookLibrary from './images/book-library.png';
import ticTacToe from './images/tic-tac-toe.png';

export const logos = [
    { name: 'javascript', logo: javascript },
    { name: 'react', logo: react },
    { name: 'vueJs', logo: vueJs },
    { name: 'nodeJs', logo: nodeJs },
    { name: 'laravel', logo: laravel },
]

export const projects = [
    { name: 'Calculator App', link: 'https://chuks-calculator.herokuapp.com/', image: calculator },
    { name: 'Book Library App', link: 'ttps://chuks-book-library.herokuapp.com/', image: bookLibrary },
    { name: 'Tic-Tac-Toe App', link: 'https://ttt-9.herokuapp.com/', image: ticTacToe },
]

export const contactLinks = [
    { name: 'Send a mail', link: 'mailto:chukkyalozie@gmail.com', icon: 'fa fa-at', target: null },
    { name: 'Call me', link: 'tel:+2348067877325', icon: 'fa fa-mobile', target: null },
    { name: 'Linkedin', link: 'https://linkedin.com/in/chukwuemekaalozie', icon: 'fa fa-linkedin-square', target: '_blank' },
    { name: 'GitHub', link: 'https://github.com/Chukky9', icon: 'fa fa-github', target: '_blank' },
]