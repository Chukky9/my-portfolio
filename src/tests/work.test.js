import { screen, render } from '@testing-library/react';
import Work from '../components/work';
import { projects } from '../data';

test('work page renders with images and links successfully', () => {
    const { container } = render(<Work/>)
    const projectImages = screen.getAllByRole('img')
    const links = screen.getAllByRole('link')

    expect(projectImages).toHaveLength(projects.length)
    expect(links).toHaveLength(projects.length)
    links.forEach(link => {
        expect(link).toBeEnabled()
    })
    expect(container).toMatchSnapshot()
})