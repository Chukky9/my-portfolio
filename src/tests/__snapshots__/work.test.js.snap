// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`work page renders with images and links successfully 1`] = `
<div>
  <section
    class="projects-section"
  >
    <h3
      class="project-section-header"
    >
      These are some of my projects
    </h3>
    <div
      class="projects"
    >
      <div
        class="project-div"
      >
        <div
          class="image-div"
        >
          <img
            alt="Calculator App"
            class="img-responsive images"
            src=""
          />
        </div>
        <div
          class="project-desc"
        >
          <a
            href="https://chuks-calculator.herokuapp.com/"
            target="_blank"
          >
            <h4>
              Calculator App
            </h4>
          </a>
        </div>
      </div>
      <div
        class="project-div"
      >
        <div
          class="image-div"
        >
          <img
            alt="Book Library App"
            class="img-responsive images"
            src=""
          />
        </div>
        <div
          class="project-desc"
        >
          <a
            href="ttps://chuks-book-library.herokuapp.com/"
            target="_blank"
          >
            <h4>
              Book Library App
            </h4>
          </a>
        </div>
      </div>
      <div
        class="project-div"
      >
        <div
          class="image-div"
        >
          <img
            alt="Tic-Tac-Toe App"
            class="img-responsive images"
            src=""
          />
        </div>
        <div
          class="project-desc"
        >
          <a
            href="https://ttt-9.herokuapp.com/"
            target="_blank"
          >
            <h4>
              Tic-Tac-Toe App
            </h4>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
